### PR TITLE
adding option to filter by metadata for boolean and integers

### PR DESCRIPTION
**Pull request by danielhuppmann**
This PR reworks the `pattern_match()'  function to allow filtering by metadata when entries in the columns are boolean or integers.

### CODE REVIEW

File: pyam_analysis/utils.py [hunk 1]
```diff
@@ -166,23 +166,26 @@ def find_depth(data, s, level):
...
-        regexp = (str(s)
-                  .replace('|', '\\|')
-                  .replace('*', '.*')
-                  .replace('+', '\+')
-                  ) + "$"
-        pattern = re.compile(regexp)
-        subset = filter(pattern.match, data)
-        depth = True if level is None else find_depth(data, s, level)
-        matches |= (data.isin(subset) & depth)
+    values = values if isinstance(values, list) else [values]
```
Comment (gidden) [inline pyam_analysis/utils.py]: here, using

```
if isinstance(values, collections.Iterable) and not utils.isstr(values)
```
would be more robust

File: tests/test_core.py [hunk 2]
```diff
@@ -265,3 +265,15 @@ def test_add_metadata_as_int(meta_df):
 
     obs = meta_df['meta_int']
     pd.testing.assert_series_equal(obs, exp)
+
+
+def test_filter_by_metadata_bool(meta_df):
+    meta_df.metadata([True, False], 'exclude')
```
Comment (gidden) [inline tests/test_core.py]: pedantic, I know, but can you add the kwarg `name='exclude'` please? I was curious what this was and went searching..

File: tests/test_core.py [hunk 3]
```diff
@@ -265,3 +265,15 @@ def test_add_metadata_as_int(meta_df):
...
+    obs = meta_df.filter({'exclude': True})
+    assert obs['scenario'].unique() == 'a_scenario'
+
+
+def test_filter_by_metadata_int(meta_df):
+    meta_df.metadata([1, 2], 'value')
```
Comment (gidden) [inline tests/test_core.py]: and same here please

**Comment by gidden** [comment]
comments in line, otherwise lgtm!

**Comment by danielhuppmann** [comment]
thanks @gidden for the comments. Should be good to go!

File: pyam_analysis/utils.py [hunk 4]
```diff
@@ -166,23 +167,29 @@ def find_depth(data, s, level):
...
-                  .replace('+', '\+')
-                  ) + "$"
-        pattern = re.compile(regexp)
-        subset = filter(pattern.match, data)
-        depth = True if level is None else find_depth(data, s, level)
-        matches |= (data.isin(subset) & depth)
+    if not isinstance(values, collections.Iterable) and not isstr(values):
+        values = [values]
+
+    values = values if isinstance(values, list) else [values]
```
Comment (gidden) [inline pyam_analysis/utils.py]: I think this can go away now, right?

File: pyam_analysis/utils.py [hunk 5]
```diff
@@ -166,23 +167,29 @@ def find_depth(data, s, level):
...
+    for s in values:
+        if isinstance(s, six.string_types):
```
Comment (gidden) [inline pyam_analysis/utils.py]: can use `isstr()` here

**Comment by gidden** [comment]
more comments, sorry!

**Comment by gidden** [comment]
looks good, though you're making my `plots` rebase a pain in the ass! =)

**Comment by danielhuppmann** [comment]
sorry about the rebase... maybe we should switch from Git to SVN?

**Comment by gidden** [comment]
hah. it was much easier than I expected anyway. we could also go back to just copying and emailing files around.

**Pull request merged**